proposals/0652-import-shadowing.rst

```diff
@@ -265,29 +265,39 @@ something like
 then the ``foo`` exported by ``A`` should be the one defined in
 ``A``'s top-level.
 
-When modules are reexported wholesale, shadowing doesn't come into
-play, and so we keep the behaviour without this extension: the form
-``module M`` names the set of all entities that are in scope with both
-an unqualified name ``e`` and a qualified name ``M.e``. Example:
+In an export list, `Paragraph 5 of Section 5.2 of the Haskell 2010 report
+<https://www.haskell.org/onlinereport/haskell2010/haskellch5.html#x11-1000005.2>`_
+specifies that the form ``module M`` names the set of all entities
+that are in scope with both an unqualified name ``e`` and a qualified
+name ``M.e``.  So in this example, *without ``ImportShadowing``*:
 
 ::
 
  module A (module M) where
 
- import M -- this exports "foo"
+ import M (foo, wombat)
+ import N (foo)
 
- foo = ...
+ foo = True
+
+the ``module M`` exports ``M.foo`` because that entity is in scope
+both as ``M.foo`` and with unqualified name ``foo``.  The fact that
+``N.foo`` and ``A.foo`` are *also* in scope with unqualified name
+``foo`` does not matter.
 
-Here, it is ``M.foo`` that is (re-)exported by ``A``, not ``A.foo``.
+With ``ImportShadowing``, however, the local definition of ``A.foo``
+*completely hides* the unqualified imports of ``M.foo`` and ``N.foo``;
+so now ``M.foo`` is no longer in scope with unqualified name ``foo``;
+so the ``module M`` export list exports only ``M.wombat``.
 
-If both ``module M`` and ``foo`` are exported, then that is a
-conflicting export error, and should be reported the same way as
-conflicts between exporting ``module M1`` and ``module M2`` without
-this extension. Example:
+If both ``M.foo`` and ``foo`` are exported, then that is a conflicting
```

Comment:
This paragraph is confusing.  It sounds as if it is related to the previous stuff about `ImportShadowing` and `module M` exports; even the use of `foo` and `M.foo` ties it to the previous example.

But it's not related at all.  Start a new subsection

## Conflicting exports

Section 5.2 of the Haskell 2010 report <link> says "The unqualified names of the entities exported by a module must all be distinct".  That condition remains unchanged with this proposal.    For example consider
```
  module A( module M, module N, foo ) where
     import M( foo, wombat )
     import N( wombat )
    foo = True
```
Without `ImportShadowing` the export list would attempt to export `M.foo` and `A.foo`; and would report a conflict.  It also attempts to export `M.wombat` and `N.wombat` and would again report a conflict (assuming they are distinct entities).
 
With `ImportShadowing` the `wombat` behaviour is unchanged; but the exports `module M` and `foo` do not conflict since `module M` no longer exports `M.foo` (see Section X above)